### DISCUSSION TITLE
I need help for improving speed of the exports...

### DISCUSSION THREAD

**Discussion by BUZZARDGTA** [Tyrrrz/DiscordChatExporter#1132] [General]
I'm using this project to do something pretty huge, I plan on exporting around 60 Discord server and retrieve all the possible available channels in export.
The goal will be to retrieve every single links sent in all of these servers. Now I'm kindda wondering, I'm at my first server attempt, and it already takes ages to export, it doesn't even reach 10% in 2 hours or something...
I tried both CLI and GUI versions, with a few parameters (I tried the only output links option that is somewhere there in the docs...)

Well  nothing seems fast. Any suggestions to improve my output being generated faster..? Cuz currently I think I would be dead before I achieve it lol

Also trough this post I want to make 2 requests:
1.1 Have the possibility to sort the servers from members (this can helps to know if the export is likely gonna take a very long time)
1.2 Have the possibility to sort the servers from messages sent (this can significantly helps to know if the export is likely gonna take a very long time)
2.1 Makes it possible to see a percentage progress in the GUI output, I believe I seen that request somewhere in the discussions there.
2.2 I think the reason I got my progress stuck in 2.1 is that my PC have hibernated automatically after idle, maybe code something that detects such things, and take whatever action you want on that.

**Comment by Tyrrrz** [comment]
Your best bet at improving speed is to use a bot token and setting parallelization to higher values. You can use message filters to only include messages with links (it will still have to pull all of them, but the output will be smaller in size). Otherwise, there's only so much you can do to speed up the process of fetching tens of millions of messages (which is usually the case on busier servers).

**Reply by BUZZARDGTA** [reply]
Thank you, I will try that soon

**Reply by BUZZARDGTA** [reply]
oh and what about my suggestions 1.1 and 1.2

**Reply by Tyrrrz** [reply]
1.1 is partially impossible (member lists are private on larger servers)
1.2 is impossible

**Reply by BUZZARDGTA** [reply]
mhmm... 1.1 will be useful.

**Reply by BUZZARDGTA** [reply]
1.2 is also possible:
https://www.reddit.com/r/discordapp/comments/5xtpfc/count_how_many_messages_in_my_server/

proof:
![image](https://github.com/Tyrrrz/DiscordChatExporter/assets/62464560/31be3b9b-724a-4508-8c68-db12dd8506f2)

**Reply by BUZZARDGTA** [reply]
please consider adding one of these at least. It would be much useful for me, and maybe others, who knows.

**Reply by Tyrrrz** [reply]
You can't access search through the API, IIRC

**Reply by qqzero0** [reply]
Yeah, only users can access this endpoint. but it would be a good feature to add to the Exporter's user mode.

**Comment by BUZZARDGTA** [comment]
`-f|--format       Export format. Choices: "PlainText", "HtmlDark", "HtmlLight", "Csv", "Json". Default: "HtmlDark".`
  
Which format would be the faster to export for the program?

**Comment by BUZZARDGTA** [comment]
another suggestion, can we export to specific folder by server name?
I don't mind not adding this feature, i'm just wondering if I mist something already existing.
